### ISSUE TITLE
Provide a list of instructions how to switch from symfony 1.4.20 to FriendsOfSymfony1 

### ISSUE THREAD

**Issue by klemens-u** [FriendsOfSymfony1/symfony1#244]
Hi everybody, and thanks for your effort to maintain symfony1.

Up until now I only cherry picked files from this repository e.g. for PHP version updates.
I now tried to switch over completely and had a few issues.

Could you please provide some information on the steps necessary to get started with 1.5.14?
Obviously some libs have been extracted.
- Which Doctrine repo do you recommend?
- Which Propel repo and version do you recommend?
- Which Swiftmailer repo and version do you recommend? Here I noticed, that the version included in 1.4.20 does not work out of the box.
- Any other repo's needed?

- There were some BC changes. What changes are usually necessary to account for them?

Thank you!

**Comment by klemens-u** [comment]
In the meantime found some information in the composer.json file. Could you please explain on how to get started with composer on an existing symfony 1.4.20 project?

**Comment by alquerci** [comment]
Hello @klemens-u 

I am going to answer all usual question that have the answer on the hand.

> Which Doctrine repo do you recommend?

https://packagist.org/packages/friendsofsymfony1/doctrine1

> There were some BC changes. What changes are usually necessary to account for them?

**Comment by fabd** [comment]
Hi guys

Thanks a lot ! I am still maintaining a symfony 1.x app.

After updating all the files I added the following to `factories.yml` as indicated in WHATS_NEW

```yaml
  mailer:
    class: sfNoMailer
```

I cleared cache with `sf cache:clear --app=the_app_name --type=config`

However my app is still trying to load SwiftMailer , I see this when I refresh the page:

```
Warning: require_once(/var/www/html/lib/vendor/symfony/lib/vendor/swiftmailer/lib/swift_required.php): failed to open stream: No such file or directory in /var/www/html/cache/koohii/dev/config/config_factories.yml.php on line 138
```

Is that normal?

Is sfNoMailer just disabling the initialization but it still requires the file? I was hoping to maybe just completely remove that folder as I don't use any of this features.

**Comment by alquerci** [comment]
Hello @fabd 

> Is that normal?

After looking for #17

I found that following.

https://github.com/FriendsOfSymfony1/symfony1/blob/1c64cbdfd98f76f0bb1d87ee5736643a32f6f627/lib/config/sfFactoryConfigHandler.class.php#L223-L227

So I can answer:
Yes, it is normal regarding the current implementation.

Solution 1
--------

A workaround is to define an empty class `Swift` on your project.

Solution 2
------

You are welcome to contribute with a PR.

cc @damienalexandre what do you think about this issue?

**Comment by damienalexandre** [comment]
I agree, SF1 require SwiftMailer classes, and sfNoMailer was only a workaround for performance (classes are loaded but not called / initialized). So you will have to keep the code even if you don't use it, or patch the ConfigHandler :+1:

**Comment by fabd** [comment]
Thank you both :+1:   I created an empty class in my appConfiguration.class.php file, just `class Swift { }`. I thought this may have been why there is a conditional in the ConfigHandler code for mailer, but I wasn't sure. Either way it works, this old project uses Zend Mail.
